contracts/plugins/assets/ERC4626FiatCollateral.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+pragma solidity 0.8.19;
+
+// solhint-disable-next-line max-line-length
+import { Asset, AppreciatingFiatCollateral, CollateralConfig, IRewardable } from "./AppreciatingFiatCollateral.sol";
+import { OracleLib } from "./OracleLib.sol";
+// solhint-disable-next-line max-line-length
+import { AggregatorV3Interface } from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import { IERC4626 } from "@openzeppelin/contracts/token/ERC20/extensions/ERC4626.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { shiftl_toFix } from "../../libraries/Fixed.sol";
+
+/**
+ * @title ERC4626FiatCollateral
+ * @notice Collateral plugin for a ERC4626 vault
```

Comment:
Should probably add a comment about how this plugin is specifically for linear 4626 vaults and does _not_ work for non-linear deposit/claim.